pdfgen/src/types/hierarchy/cross_reference_table.rs

```diff
@@ -26,15 +26,37 @@ impl CrossReferenceTable {
 
     /// Writes the contents of the `offsets`, representing them in the format required by the PDF
     /// syntax, `10 byte offset generation(00000), n`.
-    pub fn write(&self, writer: &mut impl Write) -> Result<usize, std::io::Error> {
-        let written = types::write_chain! {
+    pub fn write(&self, writer: &mut impl Write) -> Result<(), std::io::Error> {
+        types::write_chain! {
             writer.write(Self::XREF_MARKER),
             writer.write(format!("0 {}\n", self.offsets.len()).as_bytes()),
             self.offsets.iter()
                 .map(|offset| writer.write(format!("{offset:010} 00000 n{}", Self::SP_LF).as_bytes()))
                 .sum::<Result<usize, _>>(),
         };
 
-        Ok(written)
+        Ok(())
+    }
+
+    /// Returns length(size) of the `offsets` collection.
+    pub fn len(&self) -> usize {
+        self.offsets.len()
+    }
+
+    /// Returns if the `offsets` collection is empty or not.
+    #[must_use]
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Computes the 16b MD5 hash of the `offsets` collection.
+    pub fn offsets_hash(&self) -> Result<[u8; 16], std::io::Error> {
+        let bytes: Vec<u8> = self
+            .offsets
+            .iter()
+            .flat_map(|&offset| offset.to_ne_bytes())
+            .collect();
+
+        Ok(*md5::compute(&bytes))
```

Comment:
For the future we could think about using something like [md-5] crate for hashing. That crate would allow us to incrementally update the hasher (for example in the `add_object` method), and simply compute it here. 

This is absolutely fine for now though.